worker/worker.go

```diff
@@ -1,29 +1,193 @@
 package worker
 
 import (
-	"io"
+	"golang.org/x/net/context"
 
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/diff"
+	"github.com/containerd/containerd/images"
+	ctdsnapshot "github.com/containerd/containerd/snapshots"
 	"github.com/moby/buildkit/cache"
-	"golang.org/x/net/context"
+	"github.com/moby/buildkit/cache/cacheimport"
+	"github.com/moby/buildkit/cache/instructioncache"
+	"github.com/moby/buildkit/cache/metadata"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/executor"
+	"github.com/moby/buildkit/exporter"
+	imageexporter "github.com/moby/buildkit/exporter/containerimage"
+	localexporter "github.com/moby/buildkit/exporter/local"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/snapshot/blobmapping"
+	"github.com/moby/buildkit/source"
+	"github.com/moby/buildkit/source/containerimage"
+	"github.com/moby/buildkit/source/git"
+	"github.com/moby/buildkit/source/http"
+	"github.com/moby/buildkit/source/local"
+	digest "github.com/opencontainers/go-digest"
 )
 
-type Meta struct {
-	Args []string
-	Env  []string
-	User string
-	Cwd  string
-	Tty  bool
-	// DisableNetworking bool
+// WorkerOpt is specific to a worker.
+// See also CommonOpt.
+type WorkerOpt struct {
+	Name            string
+	MetadataStore   *metadata.Store
+	Executor        executor.Executor
+	BaseSnapshotter ctdsnapshot.Snapshotter // not blobmapping one
+	ContentStore    content.Store
+	Applier         diff.Differ
+	Differ          diff.Differ
+	ImageStore      images.Store
+}
+
+// CommonOpt is shared across workers
+type CommonOpt struct {
+	SessionManager *session.Manager
+}
+
+// Worker is a local worker instance with dedicated snapshotter, cache, and so on.
+// TODO: s/Worker/OpWorker/g ?
+type Worker struct {
```

Comment:
This will become interface in the future? And `containerd.NewWorkerOpt` etc will change to `containerd.NewWorker()` ? If yes, then please add `TODO` comments.